### ISSUE TITLE
Add "virtual" tag to noise channels

### ISSUE THREAD

**Issue by 95-martin-orion** [quantumlib/Cirq#2719]
Depends on #2670. Most NoiseModels today sift through the circuit and add noise channels after every moment (or operation). This becomes problematic when composing noise models (see #1835) as the second model may attempt to append noise after the noise channels from the first model.

To resolve this, these noise channels should be tagged as "virtual" (name subject to change) and all NoiseModels should skip operations with this tag when applying more noise.

**Comment by 95-martin-orion** [comment]
Initial PR is #2734.

**Comment by dabacon** [comment]
This appears to be done.  However I don't think "virtual" is the right tag.  Closing this and opening new issue for that.
